pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,24 @@
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          # normalize the path and issue 301 redirect
+          env['PATH_INFO'].gsub!('//', '/')
```

Comment:
It's a first a draft and I'm not 100% sure what I'm doing or if I'm doing the right thing :) I just wanted to get some feedback from @bryanp to see if I'm moving into the right direction or not, seems from your comment I'm a bit off ;)